src/libostree/ostree-core.c

```diff
@@ -836,6 +836,48 @@ gboolean ostree_break_hardlink (int               dfd,
   return TRUE;
 }
 
+/**
+ * ostree_fd_get_all_xattrs:
+ * @fd: File descriptor
+ * @cancellable: Cancellable
+ * @error: Error
+ *
+ * Retrieve all extended attributes in a canonical (sorted) order from
+ * the given file descriptor.
+ *
+ * Returns: (transfer full): A GVariant of type `a(ayay)`
+ */
+GVariant *
+ostree_fd_get_all_xattrs (int fd, GCancellable *cancellable, GError **error)
+{
+  GVariant *ret = NULL;
+  if (!glnx_fd_get_all_xattrs (fd, &ret, cancellable, error))
+    return NULL;
+  return ret;
+}
+
+/**
+ * ostree_get_all_xattrs_at:
+ * @fd: File descriptor
+ * @cancellable: Cancellable
+ * @error: Error
+ *
+ * Retrieve all extended attributes in a canonical (sorted) order from
+ * the given path, relative to the provided directory file descriptor.
+ * Crucically, this must be used currently to retrieve extended attributes
```

Comment:
Typo: `s/Crucically/Crucially/`

If I'm reading this correctly, I think this can be reworded in `If path is a symbolic link, it does not get dereferenced`. Although I wonder if we should future-proof this with an additional `int flags` argument.

Reply:
Addressed that except the `int flags` - not opposed to that or anything, but OTOH we're not a kernel either.  Adding new C API is much cheaper than new system calls.  I can't think what we'd ever add there that we wouldn't *also* want in the fd path?

Reply:
In the ends it's a just a matter of design decisions.
Keeping this simpler without flags and adding new APIs if needed it's fine too for me :+1: .